gems/pending/MiqVm/MiqRhevmVm.rb

```diff
@@ -4,16 +4,158 @@ class MiqRhevmVm < MiqVm
   RHEV_NFS_UID = 36
 
   def openDisks(diskFiles)
-    if @ost.nfs_mount
+    mount_storage
+    if @ost.nfs_storage_mounted
       $log.debug "MiqRhevmVm#openDisks: setting euid = #{RHEV_NFS_UID}"
       orig_uid = Process::UID.eid
       Process::UID.grant_privilege(RHEV_NFS_UID)
     end
     rv = super
-    if @ost.nfs_mount
+    if @ost.nfs_storage_mounted
       $log.debug "MiqRhevmVm#openDisks: resetting euid = #{orig_uid}"
       Process::UID.grant_privilege(orig_uid)
     end
     rv
   end
+
+  def unmount
+    super
+  ensure
+    unmount_storage
+  end
+
+  def getCfg(snap=nil)
+    cfg_props = @rhevmVm.attributes
+
+    raise MiqException::MiqVimError, "Failed to retrieve configuration information for VM" if cfg_props.nil?
+
+    storage_domains = @rhevm.storagedomains
+    $log.debug "MiqRhevmVm#getCfg: storage_domains = #{storage_domains.inspect}"
+
+    cfg_hash = {}
+    cfg_hash['displayname'] = cfg_props[:name]
+    cfg_hash['guestos']     = cfg_props.fetch_path(:os, :type)
+    cfg_hash['memsize']     = cfg_props[:memory] / 1_048_576  # in MB
+    cfg_hash['numvcpu']     = cfg_props.fetch_path(:cpu, :sockets)
+
+    # Collect disk information
+    @rhevmVm.attributes[:disks] = @rhevmVm.send(:disks, :disk) if @rhevmVm[:disks].nil?
+    @rhevmVm.disks.each_with_index do |disk, idx|
+      $log.debug "MiqRhevmVm#getCfg: disk = #{disk.inspect}"
+      storage_domain = disk[:storage_domains].first
+      storage_id     = storage_domain && storage_domain[:id]
+      storage_obj    = storage_domains_by_id[storage_id]
+
+      file_path = file_path_for_storage_type(storage_obj, disk)
+
+      tag = "scsi0:#{idx}"
+      cfg_hash["#{tag}.present"]    = "true"
+      cfg_hash["#{tag}.devicetype"] = "disk"
+      cfg_hash["#{tag}.filename"]   = file_path.to_s
+      cfg_hash["#{tag}.format"]     = disk[:format]
+    end
+    cfg_hash
+  end
+
+  def file_path_for_storage_type(storage_obj, disk)
+    storage_type = storage_obj && storage_obj.attributes[:storage][:type]
+
+    # TODO: account for Gluster and other storage types here.
+    case storage_type
+    when "nfs"
+      add_nfs_mount(storage_obj)
+      nfs_file_path(storage_obj, disk)
+    else
+      lun_file_path(storage_obj, disk)
+    end
+  end
+
+  def nfs_mount_root
+    @nfs_mount_root ||= @ost.nfs_mount_root || "/mnt/#{@rhevmVm.attributes[:id]}"
+  end
+
+  def nfs_file_path(storage_obj, disk)
+    storage_id  = storage_obj.attributes[:id]
+    disk_id     = disk.attributes[:id]
+    image_id    = disk.attributes[:image_id]
+    mount_point = nfs_mounts[storage_id][:mount_point]
+
+    ::File.join(mount_point, storage_id, 'images', disk_id, image_id)
+  end
+
+  def lun_file_path(storage_obj, disk)
+    storage_id = storage_obj.attributes[:id]
+    disk_id    = disk[:image_id].blank? ? disk.attributes[:id] : disk.attributes[:image_id]
+
+    ::File.join('/dev', storage_id, disk_id)
```

Comment:
Yes, I agree it's probably overkill here unless we pass these strings to other methods that also need to join arbitrary directories or have other needs of the Pathname class.  I noticed it because we needed to `::File` because we have our own File constant in this namespace.